app/controllers/root_controller.rb

```diff
@@ -109,12 +110,58 @@ def publication
       end
 
       @interaction_details = prepare_interaction_details(@publication, authority_slug, snac)
+    elsif @publication.format == 'local_transaction'
+      @location = fetch_location @postcode
+
+      if @location.nil?
+        @postcode_error = "fullPostcodeNoMapitMatch"
```

Comment:
There's a difference between the `nil` you get from `fetch_location` when the user hasn't entered a postcode at all and the `nil` you get from `fetch_location` when they have entered a postcode but mapit can't find it because it's not a valid postcode.  

The `else` clause further down suggests that there are interactions where we don't have to provide a postcode to proceed (because we've provided a `part` param) and so we shouldn't log a "couldn't find it in mapit" postcode error in this case.